src/services/ContributingDeveloperAuditService/scm/GitHub/GitHubAudit.ts

```diff
@@ -0,0 +1,38 @@
+import { soosLogger } from "../../../../logging";
+import { IContributingDeveloperAudit } from "../../ContributingDeveloperAuditService";
+import GitHubService from "./GitHubService";
+import { ContributingDeveloper } from "./api/GitHubApiClient";
+import { mergeContributors } from "./utilities";
+
+class GitHubAudit implements IContributingDeveloperAudit {
+  public async audit(
+    implementationParams: Record<string, string>,
+  ): Promise<ContributingDeveloper[]> {
+    const githubPAT = implementationParams["githubPAT"];
+    if (!githubPAT) {
+      throw new Error("GitHub token is required");
+    }
+    const githubService = new GitHubService(githubPAT);
+    const organizations = await githubService.getGitHubOrgs();
+    soosLogger.verboseDebug("Fetching GitHub repositories");
+    const repositories = await Promise.all(
+      organizations
+        .filter((org) => org.login === "soos-io") // TODO - REMOVE THIS FILTER ONLY FOR TESTING
+        .map((org) => githubService.getGitHubOrgRepos(org)),
+    );
+
+    soosLogger.verboseDebug("Fetching commits for each repository");
```

Comment:
The logic in here seems a bit over complicated, might dissect it a bit more / simplify it further. (there are too many maps)